scripts/morty/analysis.py

```diff
@@ -0,0 +1,273 @@
+        """
+        Used to handle concentration plotting comparisons with a single
+            method change.
+
+        Parameters
+        ----------
+        methods : list
+            List of methods to apply
+        method_name : str
+            Key for dictionary to apply method to `run_params`
+        xfactor : float
+            Value to scale `x_vals` by
+        methodname_extension : str (optional)
+            String to add to method name in plotting
+
+        Returns
+        -------
+        xs : list of list
+            Contains the `x` values to be plotted
+        ys : list of list
+            Contains the `y` values to be plotted
+        labs : list of list
+            Contains the ordered labels to use in plotting
+
+
+        """
+        xs, ys, labs = [], [], []
+        self.parasitic_results = []
+        for method in methods:
+            self.run_params[method_name] = method
+            self.data_params = DataHandler(self.run_params).data_params
+            solver = solvers.DiffEqSolvers(
+                self.run_params, self.data_params, run=False)
+            x_vals = solver.run_params['times']
+            xs.append(xfactor * x_vals)
+            result_matrix = solvers.DiffEqSolvers(
+                self.run_params, self.data_params).result_mat
+            ys.append(result_matrix)
+            labs.append(f'{method}{methodname_extension}')
+            self.parasitic_results.append(self._parasitic_abs(result_matrix))
+        return xs, ys, labs
+
+    def _save_data(self, xs, ys, labs, xlab, ylab, savename):
+        """
+        Saves the collected data into a single dictionary
+
+        Parameters
+        ----------
+        xs : list of list
+            Contains the `x` values to be plotted
+        ys : list of list
+            Contains the `y` values to be plotted
+        labs : list of list
+            Contains the ordered labels to use in plotting
+        xlab : str
+            Name of xlabel
+        ylab : str
+            Name of ylabel
+        savename : str
+            Name of file to save as
+
+
+        """
+        self.data = {}
+        self.data['xs'] = xs
+        self.data['ys'] = ys
+        self.data['labs'] = labs
+        self.data['xlab'] = xlab
+        self.data['ylab'] = ylab
+        self.data['savename'] = savename
+        self.data['parasitic'] = self.parasitic_results
+        num_nucs = []
+        for i, x in enumerate(self.data['xs']):
+            num_nucs.append(np.shape(self.data['ys'][i])[2])
+
+        num_nucs = max(num_nucs)
+        for nuclide_i in range(num_nucs):
+            final_data_points = []
+            final_names = []
+            for i, x in enumerate(self.data['xs']):
+                nuclide = self.data_params['tracked_nucs'][nuclide_i]
+                try:
+                    y = np.mean(self.data['ys'][i][:, :, nuclide_i], axis=1)
+                except IndexError:
+                    continue
```

Comment:
It might not be a bad idea to report index errors as warnings up the chain so that you know when they happened.